clarity/src/vm/database/clarity_db.rs

```diff
@@ -855,13 +856,9 @@ impl<'a> ClarityDatabase<'a> {
         Ok(())
     }
 
-    fn tenure_height_key() -> &'static str {
-        "_stx-data::tenure_height"
-    }
-
     /// Returns the tenure height of the current block.
     pub fn get_tenure_height(&mut self) -> Result<u32> {
-        self.get_data(&Self::tenure_height_key())?
+        self.get_data(TENURE_HEIGHT_KEY)?
```

Comment:
Should this function and `set_tenure_height` below be gated?

Reply:
Also, should this fall back to getting the block height if the tenure height key is not mapped in this MARF?  Thinking about what could happen if this function is reachable within a `(at-block SOME_BLOCK_BEFORE_NAKAMOTO ..)` closure.

Reply:
Ah, ok, I see what you mean. I was thinking that returning that error would be reasonable if you did something like:

```clarity
(at-block (unwrap-panic (get-block-info? id-header-hash u0)) tenure-height)
```

But maybe your idea of returning the block height would be better.

Reply:
Ah, I wasn't trying to be prescriptive here.  I was mainly interested in (1) determining if this code could lead to an `InterpreterError::Expects(..)` (or worse), and (2) if so, starting a conversation on what the right behavior ought to be (so we can add regression tests for it).

Reply:
Also, per your comment earlier, you _can_ have contract-calls within `at-block` as long as they're read-only.  So, it could be possible to call a Clarity 1 or 2 contract from a Clarity 3 function, and it would be possible to do this for either a Nakamoto block or a Stacks epoch2.x block.  I think we should determine what happens if these Clarity 1/2 contracts use `stacks-block-height`, `tenure-height`, and `block-height` when called this way (recalling that Clarity 1/2 contracts can bind these first two global variables as local names, which would now be forbidden).

Reply:
Got it. I think it does makes sense to return the block height in that situation. I've got a corresponding test ready for that. I'm a little unsure about whether the gating is necessary for `set_tenure_height`, but I can add it.